src/window/building/depot.c

```diff
@@ -0,0 +1,441 @@
+
+    text_draw(translation_for(TR_BUILDING_INFO_DEPOT_SOURCE), x_offset, y_offset + depot_order_buttons[1].y + 6, FONT_SMALL_PLAIN, 0);
+    button_border_draw(x_offset + depot_order_buttons[1].x, y_offset + depot_order_buttons[1].y,
+        depot_order_buttons[1].width, depot_order_buttons[1].height, data.focus_button_id == 2);
+    text_draw_label_and_number_centered(lang_get_string(28, src->type),
+        src->storage_id, "", x_offset + depot_order_buttons[1].x, y_offset + depot_order_buttons[1].y + 6,
+        depot_order_buttons[1].width, FONT_SMALL_PLAIN, 0);
+
+    text_draw(translation_for(TR_BUILDING_INFO_DEPOT_DESTINATION), x_offset, y_offset + depot_order_buttons[2].y + 6, FONT_SMALL_PLAIN, 0);
+    button_border_draw(x_offset + depot_order_buttons[2].x, y_offset + depot_order_buttons[2].y,
+        depot_order_buttons[2].width, depot_order_buttons[2].height, data.focus_button_id == 3);
+    text_draw_label_and_number_centered(lang_get_string(28, dst->type),
+        dst->storage_id, "", x_offset + depot_order_buttons[2].x, y_offset + depot_order_buttons[2].y + 6,
+        depot_order_buttons[2].width, FONT_SMALL_PLAIN, 0);
+}
+
+int window_building_handle_mouse_depot(const mouse *m, building_info_context *c)
+{
+    return generic_buttons_handle_mouse(m, c->x_offset + DEPOT_BUTTONS_X_OFFSET, c->y_offset + DEPOT_BUTTONS_Y_OFFSET,
+        depot_order_buttons, 5, &data.focus_button_id);
+}
+
+static void order_set_source(int index, int param2)
+{
+    window_building_info_depot_select_source();
+}
+
+static void order_set_destination(int index, int param2)
+{
+    window_building_info_depot_select_destination();
+}
+
+static void order_set_condition_type(int index, int param2)
+{
+    window_building_info_depot_toggle_condition_type();
+}
+
+static void order_set_condition_threshold(int index, int param2)
+{
+    window_building_info_depot_toggle_condition_threshold();
+}
+
+void draw_order_source_destination_background(building_info_context *c, uint8_t *title)
+{
+    int y_offset = window_building_get_vertical_offset(c, 28);
+    c->help_id = 0;
+    outer_panel_draw(c->x_offset, y_offset, 29, 28);
+    text_draw_centered(title, c->x_offset, y_offset + 10, 16 * c->width_blocks, FONT_LARGE_BLACK, 0);
+    inner_panel_draw(c->x_offset + 16, y_offset + 42, c->width_blocks - 2, 21);
+}
+
+void window_building_draw_depot_select_source(building_info_context *c)
+{
+    draw_order_source_destination_background(c, translation_for(TR_BUILDING_INFO_DEPOT_SELECT_SOURCE_TITLE));
+}
+
+void draw_depot_select_source_destination(building_info_context *c) {
+    int y_offset = window_building_get_vertical_offset(c, 28);
+
+    scrollbar.x = c->x_offset + 16 * (c->width_blocks - 2) - 26;
+    scrollbar.y = y_offset + 46;
+    scrollbar_draw(&scrollbar);
+
+    int index = 0, offset = scrollbar.scroll_position;
+
+    for (building *b = building_first_of_type(BUILDING_GRANARY); b; b = b->next_of_type) {
+        if (index >= MAX_VISIBLE_ROWS) {
```

Comment:
Don't forget to filter out invalid buildings, like deleted ones.

Reply:
Done.